src/controller/subtitle-stream-controller.js

```diff
@@ -28,186 +36,208 @@ class SubtitleStreamController extends TaskLoop {
-            frag.trackId = trackId;
-            fragQueue.push(frag);
-            this.nextFrag();
-          }
-        }
-      });
+  // If something goes wrong, procede to next frag, if we were processing one.
+  onError (data) {
+    let frag = data.frag;
+    // don't handle error not related to subtitle fragment
+    if (!frag || frag.type !== 'subtitle') {
+      return;
     }
+    this.state = State.IDLE;
   }
 
   // Got all new subtitle tracks.
   onSubtitleTracksUpdated (data) {
     logger.log('subtitle tracks updated');
+    this.tracksBuffered = [];
     this.tracks = data.subtitleTracks;
-    this.clearVttFragQueues();
-    this.vttFragSNsProcessed = {};
-    this.tracks.forEach(track => {
-      this.vttFragSNsProcessed[track.id] = [];
+    this.tracks.forEach((track) => {
+      this.tracksBuffered[track.id] = [];
     });
   }
 
   onSubtitleTrackSwitch (data) {
     this.currentTrackId = data.id;
     if (!this.tracks || this.currentTrackId === -1) {
+      this.clearInterval();
       return;
     }
 
-    // Check if track was already loaded and if so make sure we finish
-    // downloading its frags, if not all have been downloaded yet
+    // Check if track has the necessary details to load fragments
     const currentTrack = this.tracks[this.currentTrackId];
     if (currentTrack && currentTrack.details) {
-      this.tick();
+      this.setInterval(TICK_INTERVAL);
     }
   }
 
   // Got a new set of subtitle fragments.
-  onSubtitleTrackLoaded () {
-    this.tick();
+  onSubtitleTrackLoaded (data) {
+    const { id, details } = data;
+
+    if (!this.tracks) {
+      logger.warn('Can not update subtitle details, no tracks found');
+      return;
+    }
+
+    if (this.tracks[id]) {
+      logger.log('Updating subtitle track details');
+      this.tracks[id].details = details;
+    }
+
+    this.setInterval(TICK_INTERVAL);
   }
 
   onKeyLoaded () {
     if (this.state === State.KEY_LOADING) {
       this.state = State.IDLE;
-      this.tick();
     }
   }
 
   onFragLoaded (data) {
-    let fragCurrent = this.fragCurrent,
-      decryptData = data.frag.decryptdata;
-    let fragLoaded = data.frag,
-      hls = this.hls;
+    const fragCurrent = this.fragCurrent;
+    const decryptData = data.frag.decryptdata;
+    const fragLoaded = data.frag;
+    const hls = this.hls;
+
     if (this.state === State.FRAG_LOADING &&
         fragCurrent &&
         data.frag.type === 'subtitle' &&
         fragCurrent.sn === data.frag.sn) {
       // check to see if the payload needs to be decrypted
-      if ((data.payload.byteLength > 0) && (decryptData != null) && (decryptData.key != null) && (decryptData.method === 'AES-128')) {
-        let startTime;
-        try {
-          startTime = performance.now();
-        } catch (error) {
-          startTime = Date.now();
-        }
+      if ((data.payload.byteLength > 0) &&
+          (decryptData != null) && (decryptData.key != null) &&
```

Comment:
```suggestion
      if (data.payload.byteLength > 0 && (decryptData && decryptData.key && decryptData.method === 'AES-128'))
```

I would avoid double equals because of unexpected coercion behavior

Reply:
👍 that was part of the code we got from the previous PR on this

shouldn't that fail on linting?